### PR TITLE
react-big-calendar - initial typings for TimeGrid, WorkWeek, Week, Day

### PR DESCRIPTION
**Pull request by princefishthrower**
Tests add as well. Added myself as one of the authors. May need assistance from the creators / maintainers at `react-big-calendar` for more specific typings, especially the event handlers. Also, I'm not sure how "strong" the tests need to be.

Mainly, the biggest part of this pull request is adding types for [TimeGrid](https://github.com/jquense/react-big-calendar/blob/master/src/TimeGrid.js), as requested in [https://github.com/DefinitelyTyped/DefinitelyTyped/issues/48528](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/48528)

I used Lyft's [react-javascript-to-typescript-transform](https://github.com/lyft/react-javascript-to-typescript-transform) converter to help with converting TimeGrid's PropTypes into the TimeGridProps - curious to know if the `react-big-calendar` creators / maintainers have a different flow, or what tools they are using to build the types.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [X] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-big-calendar ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-big-calendar@latest/))
was missing the following properties:
1. Views
1. Navigate
1. components
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 136d066d2505f7386e9b0412abb6b8bf95eb02b9
</p>

File: types/react-big-calendar/index.d.ts [hunk 1]
```diff
@@ -392,5 +393,68 @@ export interface Views {
...
+    eventOffset: number;
+    events?: TEvent[];
+    resources?: TResource[];
+    step?: number;
+    timeslots?: number;
+    range?: any[];
+    min?: any;
+    max?: any;
+    getNow?: () => Date;
+    scrollToTime?: any;
```
Comment (martin-badin) [inline types/react-big-calendar/index.d.ts:406]: this is an instance of Date

File: types/react-big-calendar/index.d.ts [hunk 2]
```diff
@@ -392,5 +393,68 @@ export interface Views {
...
 export function move(View: ViewStatic | ViewKey, options: MoveOptions): Date;
 
+export interface TimeGridProps<TEvent extends object = Event, TResource extends object = object> {
+    eventOffset: number;
+    events?: TEvent[];
+    resources?: TResource[];
+    step?: number;
+    timeslots?: number;
+    range?: any[];
+    min?: any;
```
Comment (martin-badin) [inline types/react-big-calendar/index.d.ts:403]: this is an instance of Date

File: types/react-big-calendar/index.d.ts [hunk 3]
```diff
@@ -392,5 +393,68 @@ export interface Views {
...
+    max?: any;
```
Comment (martin-badin) [inline types/react-big-calendar/index.d.ts:404]: this is an instance of Date

**Comment by princefishthrower** [comment]
@martin-badin - thanks for those! I've added them in.

**Review by martin-badin — APPROVED** [review]

**Comment by princefishthrower** [comment]
@martin-badin - thanks for the approval! Any chance I could get this labeled with `hacktoberfest-accepted`? 😄

**Comment by princefishthrower** [comment]
Ready to merge

**Pull request merged**